Firestore/core/src/firebase/firestore/model/document_key.h

```diff
@@ -91,8 +91,9 @@ class DocumentKey {
   }
 
   /** Returns true if the document is in the specified collectionId. */
-  bool HasCollectionID(const absl::string_view collectionID) const {
-    return path().size() >= 2 && path()[path().size() - 2] == collectionID;
+  bool HasCollectionID(absl::string_view collectionID) const {
+    size_t size = path().size();
```

Comment:
Nit: this is still `collectionID`. Also, sorry I just noticed -- the function name should be `HasCollectionId`, C++ style guide mandates only capitalizing the first letter of an acronym ("_use a name like StartRpc(), not StartRPC()_").

Reply:
Oops, thanks!